packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -234,6 +239,25 @@ export const App = ({
     metricsEnabled,
   ]);
 
+  useMemo(() => {
+    // This can truthfully never be null in the current implementation,
+    // because we always generate a new one if we don't have it.
+    if (!uniqueUserId) {
+      return;
+    }
+
+    // We reuse parseAcceptLanguage with navigator.languages because
+    // that is the same as getting the headers directly as stated on MDN.
+    // See: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages
+    const [locale] = parseAcceptLanguage(navigator.languages.join(', '));
+    let [language, region] = locale.split('-');
+    if (region) {
+      region = region.toLowerCase();
+    }
+
+    initializeNimbus(uniqueUserId, { language, region } as NimbusContextT);
```

Comment:
I want to get this into AppContext or some other kind of observable state so that I can write a very simple `useExperiments` which provides a consumer with the experiment info they need in that final form that they can use anywhere in the app.

I'm not sure what to do with the results from `initializeNimbus` here.

Reply:
Maybe, I should move this to `AppContext.initializeAppContext`?

Reply:
I think this is the right direction for what we want: the experiments are fetched before the first render.

I can see this happen very early on now:

<img width="561" alt="Screenshot 2025-03-10 at 2 56 55 PM" src="https://github.com/user-attachments/assets/d63bc390-c1e1-47ec-b706-0b2516f02f4c" />

Reply:
The latest patch now has my current path forward - it doesn't seem like the worst idea since we make gql fetches at the same place, it's going to be one more request that happens now.